app/assets/javascripts/oxalis/view/action-bar/save_button.js

```diff
@@ -29,29 +37,55 @@ class SaveButton extends React.PureComponent<Props, State> {
 
   savedPollingInterval: number = 0;
   _forceUpdate = () => {
-    this.setState({ isStateSaved: Model.stateSaved() });
+    const isStateSaved = Model.stateSaved();
+    this.setState({
+      isStateSaved,
+    });
   };
 
   getSaveButtonIcon() {
     if (this.state.isStateSaved) {
       return "check";
+    } else if (this.props.isBusy) {
+      return "loading";
     } else {
       return "hourglass";
     }
   }
 
+  showProgress(): boolean {
+    // For a low action count, the progress info would show only for a very short amount of time
+    return this.props.isBusy && this.props.progressInfo.totalActionCount > 5000;
+  }
+
   render() {
+    const { progressInfo } = this.props;
     return (
       <ButtonComponent
         key="save-button"
         type="primary"
         onClick={this.props.onClick}
         icon={this.getSaveButtonIcon()}
       >
-        Save
+        {this.showProgress() ? (
```

Comment:
Minor, but would `shouldShowProgress` be a better name here as the function returns a boolean and doesn't execute an action.